src/rgw/rgw_lib.cc

```diff
@@ -524,7 +525,12 @@ namespace rgw {
     register_async_signal_handler(SIGUSR1, rgw::signal::handle_sigterm);
 
     main.init_tracepoints();
-    main.init_frontends2(this /* rgwlib */);
+    r = main.init_frontends2(this /* rgwlib */);
+    if (r != 0) {
+      derr << "ERROR: unable to initialize frontend, r = " << r << dendl;
+      exit(1);
```

Comment:
RGWLib is a library interface that allows other processes like nfs-ganesha to load rgw, so calling `exit()` is not the right way to handle errors. this error code should be returned by `RGWLib::init()`, though there's probably some cleanup we'd need to do first

Reply:
@cbodley  changing exit(1) to return -r ?

Reply:
`return r`, but call `main.shutdown()` first to clean up what we've initialized so far